subsys/nrf_compress/lzma/armthumb.c

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: 0BSD
+
+///////////////////////////////////////////////////////////////////////////////
+//
+/// \file       armthumb.c
+/// \brief      Filter for ARM-Thumb binaries
+///
+//  Authors:    Igor Pavlov
+//              Lasse Collin
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include "armthumb.h"
+
+void arm_thumb_filter(uint8_t *buf, uint32_t buf_size, uint32_t pos, bool compress)
+{
+	uint32_t i = 0;
+
+	while ((i + 4) <= buf_size) {
+		if ((buf[i + 1] & 0xF8) == 0xF0 && (buf[i + 3] & 0xF8) == 0xF8) {
```

Comment:
I know that this is same as in https://github.com/kobolabs/liblzma/blob/87b7682ce4b1c849504e2b3641cebaad62aaef87/src/liblzma/simple/armthumb.c, but is it possible that this has not been updated for Thumb 2 instruction set?
According to the armv7-m Architecture manual, the opcode, that is here 0xF8, has format:
```1 1 J1 1 J2``` so, yes it will have value 0xF8, but also 0xD0, 0xD8, 0xF0.

Reply:
Had a quick look through a disassembly of hello world and all the branch instructions are f0. On the test with the matter sample application this saved 20 or so KiB so I think the ones that are used by GCC seem to be covered well

Reply:
Yeah, that is true that compilers tend to favor some opcodes.